SCClassLibrary/Common/Files/SoundFile.sc

```diff
@@ -70,6 +71,36 @@ SoundFile {
 		^res
 	}
 
+	*write {|pathName, array, headerFormat, sampleRate, sampleFormat|
+        var file = SoundFile.new;
+
+        if(array.isKindOf(SequenceableCollection).not,
```

Comment:
I would suggest the following strategy: convert the array to a `FloatArray` at the beginning of the method. Do not check for type, but any object that won't do it, will throw an error.

Also nested arrays will be caught by this: `Array[1, [2, 2], 3].as(FloatArray)`. If errors are not informative enough, I think they should be fixed in the repective conversion (here, in FloatArray).